SpatialGDK/Source/SpatialGDK/Private/Interop/InitialOnlyFilter.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/InitialOnlyFilter.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialOSDispatcherInterface.h"
+#include "Interop/SpatialReceiver.h"
+
+DEFINE_LOG_CATEGORY(LogInitialOnlyFilter);
+
+namespace SpatialGDK
+{
+InitialOnlyFilter::InitialOnlyFilter(USpatialNetDriver* InNetDriver)
+	: NetDriver(InNetDriver)
+{
+}
+
+bool InitialOnlyFilter::HasInitialOnlyData(Worker_EntityId EntityId) const
+{
+	return (RetrievedInitialOnlyData.Find(EntityId) != nullptr);
+}
+
+bool InitialOnlyFilter::HasInitialOnlyDataOrRequest(Worker_EntityId EntityId)
+{
+	if (HasInitialOnlyData(EntityId))
+	{
+		return true;
+	}
+
+	if (InflightInitialOnlyEntities.Find(EntityId) != nullptr)
+	{
+		return false;
+	}
+
+	PendingInitialOnlyEntities.Add(EntityId);
+	return false;
+}
+
+void InitialOnlyFilter::FlushRequests()
+{
+	if (PendingInitialOnlyEntities.Num() == 0)
+	{
+		return;
+	}
+
+	TArray<Worker_Constraint> EntityConstraintArray;
+	TSet<Worker_EntityId_Key> EntitiesRequested(MoveTemp(PendingInitialOnlyEntities));
+
+	for (auto EntityId : EntitiesRequested)
+	{
+		UE_LOG(LogInitialOnlyFilter, Verbose, TEXT("Requested initial only data for entity %lld."), EntityId);
+
+		Worker_Constraint Constraints{};
+		Constraints.constraint_type = WORKER_CONSTRAINT_TYPE_ENTITY_ID;
+		Constraints.constraint.entity_id_constraint.entity_id = EntityId;
+
+		EntityConstraintArray.Add(Constraints);
+
+		InflightInitialOnlyEntities.Add(EntityId);
+	}
+
+	Worker_EntityQuery InitialOnlyQuery{};
+
+	InitialOnlyQuery.constraint.constraint_type = WORKER_CONSTRAINT_TYPE_OR;
+	InitialOnlyQuery.constraint.constraint.or_constraint.constraint_count = EntityConstraintArray.Num();
+	InitialOnlyQuery.constraint.constraint.or_constraint.constraints = EntityConstraintArray.GetData();
+	InitialOnlyQuery.snapshot_result_type_component_set_id_count = 1;
+	InitialOnlyQuery.snapshot_result_type_component_set_ids = &SpatialConstants::INITIAL_ONLY_COMPONENT_SET_ID;
+
+	const Worker_RequestId RequestID = NetDriver->Connection->SendEntityQueryRequest(&InitialOnlyQuery, SpatialGDK::RETRY_UNTIL_COMPLETE);
+	EntityQueryDelegate InitialOnlyQueryDelegate;
+	InitialOnlyQueryDelegate.BindRaw(this, &InitialOnlyFilter::HandleInitialOnlyResponse);
+
+	NetDriver->Receiver->AddEntityQueryDelegate(RequestID, InitialOnlyQueryDelegate);
```

Comment:
Is there a way we pass in worker messages from the actor system subview for this? We are actively trying to kill receiver being a catchall at the moment.